parsing/location.ml

```diff
@@ -657,14 +657,21 @@ let is_quotable_loc loc =
   && loc.loc_start.pos_fname = !input_name
   && loc.loc_end.pos_fname = !input_name
 
+let error_style () =
+  let open Misc.Error_style in
+  match !Clflags.error_style with
+  | Some Contextual | None -> Contextual
+  | Some Classic -> Classic
+
 let batch_mode_printer : report_printer =
   let pp_loc _self report ppf loc =
     let tag = match report.kind with
       | Report_error | Report_warning_as_error _ -> "error"
       | Report_warning _ -> "warning"
     in
     let highlight ppf loc =
-      if is_quotable_loc loc then
+      if error_style () = Misc.Error_style.Contextual
```

Comment:
This should be a non-fragile pattern-matching instead of an equality test, to force us to revisit if we add another option.